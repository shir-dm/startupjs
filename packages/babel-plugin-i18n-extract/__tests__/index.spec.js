const pluginTester = require('babel-plugin-tester').default
const plugin = require('../index')
const { name: pluginName } = require('../package.json')

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  pluginOptions: {},
  babelOptions: {
    plugins: ['@babel/plugin-syntax-jsx']
  },
  error: true,
  tests: [
    {
      title: 'Call without arguments',
      error: true,
      code: `
        import { observer, t } from 'startupjs'

        export default observer(function App () {
          return (
            <span>
              {t()}
            </span>
          )
        })
      `
    },
    {
      title: 'Call only with \'key\' argument',
      error: true,
      code: `
        import { observer, t } from 'startupjs'

        export default observer(function App () {
          return (
            <span>
              {t('key')}
            </span>
          )
        })
      `
    },
    {
      title: 'Call with empty \'key\' argument',
      error: true,
      code: `
        import { observer, t } from 'startupjs'

        export default observer(function App () {
          return (
            <span>
              {t('', 'defaultValue')}
            </span>
          )
        })
      `
    },
    {
      title: 'Call with empty \'defaultValue\' argument',
      error: true,
      code: `
        import { observer, t } from 'startupjs'

        export default observer(function App () {
          return (
            <span>
              {t('key', '')}
            </span>
          )
        })
      `
    },
    {
      title: 'String literal is not passed to the \'key\' argument',
      error: true,
      code: `
        import { observer, t } from 'startupjs'
        const x

        export default observer(function App () {
          return (
            <span>
              {t(x, 'defaultValue')}
            </span>
          )
        })
      `
    },
    {
      title: 'String literal is not passed to the \'defaultValue\' argument',
      error: true,
      code: `
        import { observer, t } from 'startupjs'
        const x

        export default observer(function App () {
          return (
            <span>
              {t('key', x)}
            </span>
          )
        })
      `
    },
    {
      title: 'Correct call',
      code: `
        import { observer, t } from 'startupjs'

        export default observer(function App () {
          return (
            <span>
              {t('key', 'defaultValue')}
            </span>
          )
        })
      `
    }
  ]
})
