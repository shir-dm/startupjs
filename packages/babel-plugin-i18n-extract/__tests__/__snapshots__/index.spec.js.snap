// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@startupjs/babel-plugin-i18n-extract Call only with 'key' argument: Call only with 'key' argument 1`] = `

import { observer, t } from 'startupjs'

export default observer(function App () {
  return (
    <span>
      {t('key')}
    </span>
  )
})

      â†“ â†“ â†“ â†“ â†“ â†“

Error: unknown: [@startupjs/babel-plugin-i18n-extract]: Not enough arguments passed to the function t

`;

exports[`@startupjs/babel-plugin-i18n-extract Call with empty 'defaultValue' argument: Call with empty 'defaultValue' argument 1`] = `

import { observer, t } from 'startupjs'

export default observer(function App () {
  return (
    <span>
      {t('key', '')}
    </span>
  )
})

      â†“ â†“ â†“ â†“ â†“ â†“

Error: unknown: [@startupjs/babel-plugin-i18n-extract]: Argument 'defaultValue' of t cannot be an empty string

`;

exports[`@startupjs/babel-plugin-i18n-extract Call with empty 'key' argument: Call with empty 'key' argument 1`] = `

import { observer, t } from 'startupjs'

export default observer(function App () {
  return (
    <span>
      {t('', 'defaultValue')}
    </span>
  )
})

      â†“ â†“ â†“ â†“ â†“ â†“

Error: unknown: [@startupjs/babel-plugin-i18n-extract]: Argument 'key' of t cannot be an empty string

`;

exports[`@startupjs/babel-plugin-i18n-extract Call without arguments: Call without arguments 1`] = `

import { observer, t } from 'startupjs'

export default observer(function App () {
  return (
    <span>
      {t()}
    </span>
  )
})

      â†“ â†“ â†“ â†“ â†“ â†“

Error: unknown: [@startupjs/babel-plugin-i18n-extract]: Not enough arguments passed to the function t

`;

exports[`@startupjs/babel-plugin-i18n-extract String literal is not passed to the 'defaultValue' argument: String literal is not passed to the 'defaultValue' argument 1`] = `

import { observer, t } from 'startupjs'
const x

export default observer(function App () {
  return (
    <span>
      {t('key', x)}
    </span>
  )
})

      â†“ â†“ â†“ â†“ â†“ â†“

SyntaxError: unknown: Const declarations require an initialization value (2:7)

[0m [90m 1 |[39m [36mimport[39m { observer[33m,[39m t } [36mfrom[39m [32m'startupjs'[39m[0m
[0m[31m[1m>[22m[39m[90m 2 |[39m [36mconst[39m x[0m
[0m [90m   |[39m        [31m[1m^[22m[39m[0m
[0m [90m 3 |[39m[0m
[0m [90m 4 |[39m [36mexport[39m [36mdefault[39m observer([36mfunction[39m [33mApp[39m () {[0m
[0m [90m 5 |[39m   [36mreturn[39m ([0m

`;

exports[`@startupjs/babel-plugin-i18n-extract String literal is not passed to the 'key' argument: String literal is not passed to the 'key' argument 1`] = `

import { observer, t } from 'startupjs'
const x

export default observer(function App () {
  return (
    <span>
      {t(x, 'defaultValue')}
    </span>
  )
})

      â†“ â†“ â†“ â†“ â†“ â†“

SyntaxError: unknown: Const declarations require an initialization value (2:7)

[0m [90m 1 |[39m [36mimport[39m { observer[33m,[39m t } [36mfrom[39m [32m'startupjs'[39m[0m
[0m[31m[1m>[22m[39m[90m 2 |[39m [36mconst[39m x[0m
[0m [90m   |[39m        [31m[1m^[22m[39m[0m
[0m [90m 3 |[39m[0m
[0m [90m 4 |[39m [36mexport[39m [36mdefault[39m observer([36mfunction[39m [33mApp[39m () {[0m
[0m [90m 5 |[39m   [36mreturn[39m ([0m

`;
